.dagger/build/sdk.go

```diff
@@ -86,9 +95,22 @@ func (build *Builder) typescriptSDKContent(ctx context.Context) (*sdkContent, er
 		},
 	})
 
+	sdkNodeModules := dag.Container().
+		From(tsdistconsts.DefaultNodeImageRef).
+		WithWorkdir("/work").
+		WithDirectory("/work/sdk", rootfs).
+		WithoutEntrypoint().
+		WithMountedCache("/root/.npm", dag.CacheVolume(fmt.Sprintf("npm-cache-node-%s", tsdistconsts.DefaultNodeVersion))).
```

Comment:
nit: do we use an npm cache volume anywhere else in our modules like this? If so because of module-based cache volume namespacing it would be efficient-er to define a function in one module that returns a consistent cache volume to use, so then they are all shared

This is not a big deal at all, just an idea if quick to implement

Reply:
does the typescript SDK count as a module in this case? otherwise we do not, but the SDK not only uses this for its own purposes, it also mounts this for user modules.

Reply:
How fast do we get thanks to this bundling? I saw 7s faster but it should be much more normally since installing deps always take around +10seconds.

Reply:
there are 2 bundlings: tsx saves 7 seconds pretty consistently. node modules saves 0-2s on `corepack use yarn` because irl some package versions don't end up matching and it takes a second to re-compute integrity across package managers.

Reply:
if im revisiting this to fix the extraneous go.mod i'll double check these numbers.

Reply:
yeah, revisited, the node_modules caching adds no speedup here, so theoretically slower because of the increased image size. I'm tearing it out of this PR, and will take another shot at it in an isolated setup.

exporting from DMs, too, I may've found that one of the cache volumes doesn't hit, but fixing it will require x-package-manager research:

ts sdk:
```
WithMountedCache("/root/.cache/yarn", dag.CacheVolume(fmt.Sprintf("yarn-cache-%s-%s", runtime, version))).
````
shelling in there:
```
dagger /src/.dagger $ yarn cache dir
/usr/local/share/.cache/yarn/v6
```